### ISSUE TITLE
Subscribe to GitHub Discussions 

### ISSUE THREAD

**Issue by klzns** [integrations/slack#1026]
With the new GitHub Discussions feature, it would be great if we had visibility of activities of Discussions in Slack.

**Comment by jules2689** [comment]
Thanks so much for this feature request!

We've passed it along to the discussions team who is still working on all the details for that feature as it's still in beta.
We don't have anything to report back right now, other than to thank you for opening this ❤️

**Comment by klzns** [comment]
Any news about this?

**Comment by despreston** [comment]
+1

**Comment by amogkam** [comment]
+1, this would be super useful!

**Comment by vinckr** [comment]
+1, this is still not possible right?

**Comment by maxim-lobanov** [comment]
Is there any ETA on this feature? It would be super useful!

**Comment by Ngoguey42** [comment]
+1

**Comment by vinckr** [comment]
Looking at https://github.com/integrations/slack/issues/1103 I doubt this will ever happen. 
This will probably be easy to pull off when there is a proper Github API endpoint for the discussions.

**Comment by eregon** [comment]
I'd like to see this as well. GitHub Discussions is becoming more widely used now, and this would be a great way to gets updates on Slack for GitHub Discussions.

**Comment by simoncent** [comment]
+1, would find very useful

**Comment by becca** [comment]
@evi-liu will have some updates for y'all on this! ✨

**Comment by evi-liu** [comment]
Hey folks, Discussions PM here. Thanks for all the feedback on this so far! Our team is currently hard at work on implementing webhooks right now, so you'll be able to start integrating with Slack in the coming weeks. I'll keep this post updated once we have a private beta available to try out.

**Comment by Kaspik** [comment]
@evi-liu Hey Evi! Discussions webhook is in [public beta](https://github.blog/changelog/2021-03-30-github-discussions-webhooks-public-beta/) now - do we have ETA on final release and ETA when Slack integration might support Discussions then? :) Thanks!

**Comment by dprada** [comment]
+1, I would find it very useful

**Comment by dito** [comment]
+1

**Comment by evi-liu** [comment]
Hi folks, we're slotted to begin this work in the coming weeks, so stay tuned and I'll post an update once the Slack integration is available for testing. In the meantime, I'd love to know: what kind of Discussions info would be valuable to integrate with Slack?

**Comment by Kaspik** [comment]
- new discussion/topic opened
- discussion marked as answered

I'm not sure about comments - it could become pretty spammy. But without notification about "new discussion opened" it's pretty easy to miss a new topic.

**Comment by stephenyeargin** [comment]
*Idea:* Let's move the discussions (on discussions, how meta!) to https://github.com/integrations/slack/discussions/1140 since there's an open thread there.

**Comment by dbeckwith** [comment]
I agree that comment notifications could be spammy but would still like the option to have those for verbose slack channels and so people can follow along with discussions without needing to open GitHub. A distinct notification for "discussion marked as answered" sounds very useful too.

**Comment by barkerja** [comment]
> In the meantime, I'd love to know: what kind of Discussions info would be valuable to integrate with Slack?

1. Discussion Creation
2. Personal replies (mentions)
3. Discussion closed (answered)

**Comment by alrightjones** [comment]
> Hi folks, we're slotted to begin this work in the coming weeks, so stay tuned and I'll post an update once the Slack integration is available for testing.

@evi-liu has the slack app <> discussions integration baked or still waiting on testing? Thanks in advance🙏

**Comment by evi-liu** [comment]
still actively being worked on our team! sorry, it's been taking a while. we've run into some unexpected complications, but hope to start beta testing with folks very, very soon 🙏

**Comment by alrightjones** [comment]
@evi-liu we'd love to be a beta test partner! We have a large active community in slack (2.5k) and we need to start moving the discussions from Slack >> GitHub Discussions. This would be 👩🏾‍🍳 💋 !

**Comment by vinckr** [comment]
Would also love to beta test 😄

**Comment by peteroneilljr** [comment]
Also happy to test this if anyone has a beta available!

**Comment by sryza** [comment]
+1

**Comment by JasonTheAdams** [comment]
Would love to be in on that beta! Pretty easy to miss important messages right now. 😬

**Comment by antn** [comment]
Hey folks! Thanks for your patience. This is now available on Slack! 🎉 

You can try it out with `/github subscribe owner/repo discussions`.

You'll receive notifications whenever a discussion is created or when a discussion has a comment [marked as an answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer).

This feature requires granting the Slack app `read` permissions for discussions in your organization. If necessary, you should be prompted to update the app permissions when subscribing to discussions. Please note that you must be an organization owner to approve this permissions update.

Let us know if you have any feedback!

**Comment by jeffpaul** [comment]
@antn I'm seeing that using `/github subscribe owner/repo discussions` also adds on other things like commits, etc when just discussions are expected.  Is that a known bug?

**Comment by antn** [comment]
@jeffpaul ah, sorry for the confusion. When adding a [new subscription to a repository and specifying a feature](https://github.com/integrations/slack#configuration), it enables subscriptions for our default features, _and_ the feature you specify in the command.

If you want to only receive notifications for discussions, you can unsubscribe from those default subscriptions:

`/github unsubscribe owner/repo issues pulls commits releases deployments`

Hope that helps!

**Comment by creasty** [comment]
Thank you, this is really helpful!

> You'll receive notifications whenever a discussion is created or when a discussion has a comment [marked as an answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer).

@antn Could you clarify the last part?
Does this mean regular comments won't be notified?
Ones before being marked as an answer and any comments under a category that have answers feature disabled.

**Comment by antn** [comment]
@creasty Subscribing to discussions comments isn't supported yet, that's correct. We're open to adding this in the future, but this initial feature only notifies when a discussion comment is marked as an answer.

**Comment by gelicia** [comment]
I work on a repo that funnels feature requests into discussions. We would love to be able to be notified when a discussion for our team (signified by the discussion's label) is created, or when the discussion is edited to include our label.

**Comment by blakewilson** [comment]
@antn Thanks a lot for the discussions integration. This will be really helpful for our team. Is there any roadmap to support discussion comment notifications as well?

**Comment by rowi1de** [comment]
Is it possible to filter Discussions by category?

**Comment by lukens** [comment]
> @creasty Subscribing to discussions comments isn't supported yet, that's correct. We're open to adding this in the future, but this initial feature only notifies when a discussion comment is marked as an answer.

I would really love if discussion comments could be subscribed to too, like they can for issues (including with the `channel` option to broadcast them to the channel, as well as posting them in a thread on the parent).
